### ISSUE TITLE
TensorflowLite model run on Hexagon DSP different from CPU

### ISSUE THREAD

**Issue by GinHIT** [tensorflow/tensorflow#56848]
Hi, I convert a "resnet-like" .tflite model(quantized by int8) and find it has a certain precision loss on hexagon compared to cpu,  then i print the output probability,  they are really different.
I used to discover the "inference_diff" tool (https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/tools/evaluation/tasks/inference_diff) that you guys provided and now test with it, the result is as follows:
---------------------------------------------------------------------------------------------------------------------------
```
./run_eval --model_file=resnet_quantized.tflite --delegate=hexagon                                                                                             
INFO: Initialized TensorFlow Lite runtime.
loaded libcdsprpc.so
Hexagon delegate created.
INFO: TfLiteHexagonDelegate delegate: 34 nodes delegated out of 36 nodes with 1 partitions.
VERBOSE: Replacing 34 node(s) with delegate (TfLiteHexagonDelegate) node, yielding 3 partitions.
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
native : lite/tools/evaluation/stages/inference_profiler_stage.cc:78 Test interpreter has been initialized.
native : lite/tools/evaluation/stages/tflite_inference_stage.cc:146 
native : lite/tools/evaluation/stages/inference_profiler_stage.cc:92 Reference interpreter (1 thread on CPU) has been initialized.
Num evaluation runs: 50
Reference run latency: avg=240374(us), std_dev=142248(us)
Test run latency: avg=19779(us), std_dev=5008(us)
OutputDiff[0]: avg_error=0.000274915, std_dev=4.89316e-05
```
----------------------------------------------------------------------------------------------------------------------------
It looks like little difference, but in general, the output probability of the quantized model are mostly 0, so this probably doesn't mean much. Then I remove the softmax operation at the end of the model, and try again:
----------------------------------------------------------------------------------------------------------------------------
```
./run_eval --model_file=resnet_quantized_without_softmax.tflite --delegate=hexagon                                                                             
INFO: Initialized TensorFlow Lite runtime.
loaded libcdsprpc.so
Hexagon delegate created.
INFO: TfLiteHexagonDelegate delegate: 33 nodes delegated out of 35 nodes with 1 partitions.
VERBOSE: Replacing 33 node(s) with delegate (TfLiteHexagonDelegate) node, yielding 3 partitions.
INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
native : lite/tools/evaluation/stages/inference_profiler_stage.cc:78 Test interpreter has been initialized.
native : lite/tools/evaluation/stages/tflite_inference_stage.cc:146 
native : lite/tools/evaluation/stages/inference_profiler_stage.cc:92 Reference interpreter (1 thread on CPU) has been initialized.
Num evaluation runs: 50
Reference run latency: avg=234793(us), std_dev=146766(us)
Test run latency: avg=19949.9(us), std_dev=5333(us)
OutputDiff[0]: avg_error=2.66955, std_dev=0.73623
```
----------------------------------------------------------------------------------------------------------------------------
Now the difference is much bigger. I wonder whether these are normal, and what is the approximate level of accuracy loss of the model generally. I upload these two models which named resnet_quantized.tflite.tar.gz and resnet_quantized_without_softmax.tflite.
Thanks [!](url)
[resnet_quantized.tflite.tar.gz](https://github.com/tensorflow/tensorflow/files/9157839/resnet_quantized.tflite.tar.gz)
[resnet_quantized_without_softmax.tflite.tar.gz](https://github.com/tensorflow/tensorflow/files/9157842/resnet_quantized_without_softmax.tflite.tar.gz)

**Comment by mohantym** [comment]
Hi @GinHIT !
resnet_quantized lite model is expected to have faster inference time /lower latency compared to resnet_quantized_without_softmax.tflite. Are you saying the for both models results different behavior for CPU and DSP hexagaon delegate respectively.

Could you provide the benchmarking results for both model with CPU and DSP GPU delegate . Attached relevant [thread ](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/tools/benchmark)for reference.

Thank you!

**Comment by mohantym** [comment]
@sachinprasadhs !
Could you please look at this issue.
Thank you!

**Comment by Janus-Shiau** [comment]
In our case, some building block, e.g. the residual-like conv. building block, may cause standalone ReLU operator in TFLite.
When this kind of operators occurs in graph, the results on CPU and Hexagon are usually different. 

You can also check out #54481.

**Comment by pjpratik** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFLite team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the TFLite space.

Thanks.
